### ISSUE TITLE
Can't format fragment of source code with #if element, and keyword, and triple-slash comment

### ISSUE THREAD

**Issue by knocte** [fsprojects/fantomas#2360]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAdAdgWwQVwBtEwAFAJwEtsKAXCgN0QGcwBeLCCGgTwAdiAQgVjQA1gAlEFAOYALGmw6dlAHzBCREqXIWwAZmDwVMNAMwAmMFiUQA9PbAAqRwBFm0SrzqxMzm2Htbf0CwAEYANgAjWjAyRAIAQzpGMEjhMTBZbXlDJkR4MD1YMjAAAwB5EgB9AGEAZQA1UrB00SYAGmCHOuNoYkraxrAEgmEAdzAfAm5U1rBMPGxIxBL9MAAGMAA-MHCAVj3TPfawGIoWJgS9RC7bMBpYWOYaSmgFYwf7rNiEzGliJjYEZEEo0WS-dStSQyeT%2BX7wLAAYgoBgAhAA5WAuACqTAAgkw6i88G8mP5lABtAA8RLIJJoAD4ALpIxCYeAo8mcDRiaE6cp6PR5GgRRSYZQSzhqHlaGE0AVCxAi8KTAxGEwRayYSDtcBQYV0P5MVBgCl6iCgSWQAAkPH4JsgHx15s4NvoIzwyDQZslEHWut9UAAkuy2TQegAvZAB30AFhdECZLoAvjGLQmoLa%2BF6oE6067re6CJ6TT7faF88pIABZBIADwAMsZEA22dIwc7A6FzOsM0nlKmXZaJTa7TnIGz4AKm5hEETuEROyPCx6c2XJeZK66AKLs6fNpdWgh6SB9lP54dVrP2tCQSKwWAEQ8Fosl70ZiCmLcQSAhvJkGgADFjBGdFEDGAgD2-e46UQM8BwvDNR2zB170fZ8fxXYs1w-MBY2gyA6l4BI%2BgERAijiEgEjIBJsCVFYMM4Yk4MlftOEHZRLwLMdUIfJ8tzdVdS1w45cMI4jSPI4oW1gMYVmgBI8hDegRCSCgfEYrhYPg9jEKta9xzQ-ikKwt9TVw8ICKIkjEDIijEGxXh%2BDIBSlMwFSFO8bVoL0EY8h0iAOM4LjMJ428jMYwTsOEwMAHYrIk2ypLiGpEiYJgah8JhiTeYpNMKPyWIlNjAr05cwqgCKBNMnDAwADgSmy7Ok6tECWBifMKgKwCC9N9Iqu8%2BMimqYt9ABORrJPszLhG83DfIIfzWPPIcTIGqqTNfWry39MTrL6PE9BoFZMuwIF8uY7rerAELM3Wobqq20bJVCCs9sS5q4jqNqKDgWb8oWpbipWzi1pQ8KHs2oT3y7Td3psw7jrIb7qD%2BjToMu5aENW-rwcqyH9KemHyy-eGDrIWA8HZNxIOoJGLu0rHdJx8q8cdExhqJ8yu3wsTazrIM9AAFSyTA6lkYoaAAdQoeAO2g3smdKlmrwGvMoei4mXtEwMa3rQWRbZbdFrnCWAJluXZHy2NFeB7HQdxm9cw5x7oe58tLL5-XMD0Cg63KZykmKbc614OJ0vUubAz2W3lBKnqytVtn1cJt31wlUJ4q9usACVEDgMh4At%2BXcJtq7E%2B45OXY1sz0%2BUUIGuzvOC-gdFFmWMgBSDY7sGNaDQnLlXK6dyBsEIOhINnQDiiBGgkaFlDXc192XomsTm%2BKeBq3HihJ8QaeyFn%2BmCOgcEaLeFYqjGWWO0Hh3WZHlPly5uvOB7Aj%2BbxMgaO4coyCbbKxcrbQTLkrBOQ9QpVxoJzNOuFzBvV1p-b%2BCRf7-3ODQNu7VO56G7m1PuuEB5gOurdZCI8x4EAns2A%2BR8VgL3tEvWucC4a6y-j-P%2BACaDbwobvKhM8kjHzEqfaiJEkZXxvlbO%2BwUwaP2rqnZer8IDmFJog%2BsDRVwCGMByP4QD8p1VjpweOxDpHjifleF%2BcDeYqLrIBKmbxI4aPZMYaQOiQH6MTCDKRjsTGyOfrAwM5gda%2Bj1nWFwsAaAAHElQhzDswJgkcXEiTcWAQxFdIEjw2nIxh-jPaIJ3nvGUAhz6iCVEwcoYtaKIBmosKOvpAZFTjh4vqD9DIE18fIuBWddZgQgs2MiNA5JsjoYgNwvtMC0EjnidkrUsH4MDHUyRTSk7pNaWYvxvpzCNxYZBaQmAbGYDsT4HoOykh4DiELWAIYkAmDUujeaXUiGpLumzDJbSsnrLXlspGmA1KMAbD4aQ0yO4jOMOMrKAN7l22ZvfJZLT0IMO2pKUwu1cncJnC2WikR4AJFSrAOJfxul73BSbBZN1jG8ThTXBFEpTAIKCdElYVAwwjAANKIEQLwS5YYQyFN%2BKfIlQMGn208c08lxlMlUuUBYAiQhfiiDRfiCmVNW7PHyFwyhs5onhziWCjGjNIXK2hcPWFYrXkSs4KYZRQSBDUU%2BsM84HgqAgWOvAbEYyfBuGgIkGiXl%2BX1IMY00lXjRUwPaYGUwli6WhwZXRa5BBaSU2ynSXg85FydWJQ8iBTyZHQPhc9algSrSsvZfzTBHcBQyswHK5ssydpJJSZm0hxqQ1vMRTkoJtJfqcIQNGO56b9U9T1EyGMjomCATiSaeZyYgA)

#### Code

```fsharp

module Primitives =
    type BlockHeight =
        | BlockHeight of uint32 

    /// **Description**
    ///
    /// 16bit relative block height used for `OP_CSV` locks,
    /// Since OP_CSV allow only block number of 0 ~ 65535, it is safe
    /// to restrict into the range smaller than BlockHeight
    and
#if !NoDUsAsStructs
        [<Struct>]
#endif
        BlockHeightOffset16 =
            | BlockHeightOffset16 of uint16


```
            

#### Error

```fsharp
Fantomas.Core.FormatConfig+FormatException: Fantomas is trying to format the input multiple times due to the detect of multiple defines.
There is a problem with merging all the code back together.
[] has 2 fragments
[NoDUsAsStructs] has 3 fragments
Please raise an issue at https://fsprojects.github.io/fantomas-tools/#/fantomas/preview.
   at Fantomas.Core.CodeFormatterImpl.format@87-5.Invoke(FSharpList`1 results) in /_//src/Fantomas.Core/CodeFormatterImpl.fs:line 108
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvokeNoHijackCheck[a,b](AsyncActivation`1 ctxt, b result1, FSharpFunc`2 userCode) in D:\a\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 465
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in D:\a\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 104
```
            
#### Problem description

Please describe here the Fantomas problem you encountered.
                    Check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#bug-reports).

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-07-11T16:44:40Z - 98a7449918c2167c0478a151ec89bbde826a74c0

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

**Comment by nojaf** [comment]
This indeed is a bug, the problem is that the trivia between `and` and the `[<Struct>]` isn't being restored correctly. This can be better spotted when changing it to [a comment](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAdAdgWwQVwBtEwAFAJwEtsKAXCgN0QGcwBeLCCGgTwAdiAQgVjQA1gAlEFAOYALGmw6dlAHzBCREqXIWwAZmDwVMNAMwAmMFiUQA9PbAAqRwBFm0SrzqxMzm2Htbf0CwAEYANgAjWjAyRAIAQzpGMEjhMTBZbXlDJkR4MD1YMjAAAwB5EgB9AGEAZQA1UrB00SYAGmCHOuNoYkraxrAEgmEAdzAfAm5U1rBMPGxIxBL9MAAGMAA-MHCAVj3TPfawGIoWJgS9RC7bMBpYWOYaSmgFYwf7rNiEzGliJjYEZEEo0WS-dStSQyeT%2BX7wLD2ADEFAMAEIAHKwFwAVSYAEEmHUXng3kx-MoANoAHmJZFJNAAfABdRG2JGITDwVEUzgaMTQnTlPR6PI0CKKTDKaWcNT8rQwmjC0WIcXhSYGIwmCLWTCQdrgKBiuh-JioMCUw0QUAyyAAEh4-HNkA%2B%2BqtnHt9BGeGQaEtMog6wNAagAEkuZyaD0AF7IYMBgAs7ogzPdAF949bk1AHXxfVBXZmPXavQQfeb-QHQkXlJAALIJAAeABljIhm5zpGC3SHQuZ1tnU8oM%2B6bdL7Y785BOfBha3MIhidwiD3xyXvfnKzLzDWPQBRLlztur20EPSQQfpotj2u5p1oSCRWCwAgn4ul8t%2B7MQUy7iCQcM8jIGgADFjBGDFEDGAhjz-e56UQS9h2vbMJzzZ0nxfN9-3XMtN2-MAEzgyA6l4BI%2BgERAijiEgEjIBJsFVFZsM4ElEJlIdOBHZQb2LScMOfV9d09DcKwI44CJIsiKKo4p21gMYVmgBI8nDegRCSCgfBYrgEKQriUNtO8p0woTUNwz8LQI8JiNI8jEEo6jEBxXh%2BDIZTVMwdTlO8PU4L0EY8n0iBuM4XicP4h9TJYkS8LEkMAHZbOkhzZLiGpEiYJgah8JgSTeYodMKQL2OlTiQsMtdIqgaLhIs-CQwADmS%2BzHLkutECWZj-JK4KwFCrMjOqx9BJi%2Br4oDABOFqZKcnLhD8giAoIIKOKvUdzOG2rzI-BqqyDSS7L6fE9BoFYcuwIEirYvqBrAcKcy20a6t2iaZVCatDpStq4jqTqKDgBaiuW1ayvWnjNvQqLnp20Sv17Hcvvsk6zrIP7qEB7S4JutbkI2oaoZqmGjNe%2BGq1-JHjrIWA8C5NwYOoVHrr03GDPxqrCZdEwxtJqzeyIySG0bUM9AAFSyTA6lkYoaAAdQoeBuzggdWYq9nb2GwtYbisn3okkN6ybEXxc5PcVsXaXgPlxXZCKhMVbBvGIYJ%2B8C25l64b5qsbMFo3MD0ChG3KNykmKPdG14OIsq0xaQz2B3lHK-rKo1zmtZJz2t2lUIkt9xsACVEDgMh4GtpWCPt26U74tP3e1yys%2BUUJmrzwvi-gDFFmWMhhVDM7sDNODQir9Wa9dyBsEIOgYIXEDiiBGhUdF9CPZ1r33umyS2%2BKeA6ynigZ8QOeyAXpniOgcF6LeFYqjGBXuxH52OfH9O115xvOH7YihfxMh6O4coZBWx5TLrbOCldVbJ1HhFWuNAeaZwIuYT6Bsf5-wSAAoB5waCdy6j3PQfdOqDwIsPSBd0HpoXHpPAg082zH1PisZeTpV4N0QYjA2v9-6AOATQPe1CD60PnkkM%2BkkL50XIqjW%2B99baPzCpDF%2BdcM5rw-hAcwFMUFNgaBuAQxhuR-FAUVRqCdOBJzIXIqcr9bzv0QQLdRjYQK0zeDHbRXJjDSH0eAoxKZwayJduYhRb8EEhnMPrAMhtGwuFgDQAA4qqcOkdmBMBju48SniwAmOrjA8e21FEsKCT7FB%2B9D7ygEFfUQqomDlElgxRA81FixwDCDUqidvGDWfiZYmASlGINzgbSC0E2yURoIpTkjDEBuADpgWgMd8Rcg6rgohIZGkyNaanLJHTLGBIDOYFu7CYLSEwPYzAjifA9H2UkPAcRRawHDEgEwmksZLV6qQjJj1ObZM6bkrZm9dmo0wJpRgzYfDSDmd3cZxgpm5WBk8x2bMn6rPaVhZhe0ZSmAOgUvh852wMUiPABIGVYCJL%2BH0w%2BULzbLPumYgSiL67IulKYZBoS4krCoJGEYABpRAiBeA3MjOGEpvwL6ktBs0p2Pi2lUrMjk2lygLDESEL8UQmKCTU1ph3Z4%2BReE0IXHEqOiTIXYxZjCtWcKx4IslR86VnBTBqNCQIOiP0xnnA8FQcCZ14A4kmT4Nw0BEj0V8kKppxiWkUt8RK%2BBXSQymBsYyiOzLGJ3IIHSGmeV6S8CXCuHqZLnnQNefIuBSK3p0pCbaDlXKhY4O7sKeVmBFVtgWftVJ6Sc0ULNeGz5KL8mhLpADHhCA4yPKzUa-qhpmTxhdEwECiTzRLLTEAA).

In order to solve this Fantomas needs to know about the `and` keyword in the AST.
This should be added to the [SynTypeDefnTrivia](https://fsharp.github.io/fsharp-compiler-docs/reference/fsharp-compiler-syntaxtrivia-syntypedefntrivia.html) over at `dotnet/fsharp`.

I can provide some pointers if anybody want to contribute this to the F# compiler.

**Comment by nojaf** [comment]
Fix available in [v5.2.0-beta-001](https://www.nuget.org/packages/fantomas/5.2.0-beta-001)
